.github/workflows/publish.yaml

```diff
@@ -1,39 +1,52 @@
 name: CI
 
-on: [pull_request, push]
+on: 
+  pull_request:
+  push:
+    branches: [master]
 
 jobs:
   publish:
     runs-on: ubuntu-latest
-
     steps:
-      - name: Checkout Snapcraft
+      - name: Decision to Publish
+        id: decisions
+        run: |
+          # Secrets cannot be used in conditionals, so this is our dance:
+          # https://github.com/actions/runner/issues/520
+          if [[ -n "${{ secrets.STORE_LOGIN }}" ]]; then
+            echo "::set-output name=PUBLISH::true"
+          else
+            echo "::set-output name=PUBLISH::false"
+          fi
+
+          if [[ ${{ github.event_name }} == 'pull_request' ]]; then
+            echo "::set-output name=PUBLISH_BRANCH::edge/pr-${{ github.event.number }}"
+          elif [[ "${GITHUB_REF##*/}" == "master" ]]; then
+            echo "::set-output name=PUBLISH_BRANCH::edge"
```

Comment:
I don't think we want this, PUBLISH::false!

Reply:
I assume you don't like the publish branch being set?

The steps are always conditional on PUBLISH, not PUBLISH_BRANCH FWIW

Reply:
I don't want accidents :-)